src/GraphMsg.chpl

```diff
@@ -0,0 +1,4510 @@
+                  forall i in srcR.localSubdomain(){
+                        srcR[i]=dst[i];
+                        dstR[i]=src[i];
+                   }
+               }
+             }
+             combine_sortR();
+             neighbourR=0;
+             start_iR=-1;
+             set_neighbourR();
+
+          }
+          if (RCMFlag>0) {
+             RCM_u();
+             neighbour=0;
+             start_i=-1;
+             combine_sort();
+             set_neighbour();
+             coforall loc in Locales  {
+                  on loc {
+                      forall i in srcR.localSubdomain(){
+                            srcR[i]=dst[i];
+                            dstR[i]=src[i];
+                       }
+                  }
+              }
+              neighbourR=0;
+              start_iR=-1;
+              combine_sortR();
+              set_neighbourR();
+
+          }   
+      }//end of undirected
+      else {
+
+        if (RCMFlag>0) {
+           RCM();
+           neighbour=0;
+           start_i=-1;
+           combine_sort();
+           set_neighbour();
+
+        }
+
+      }
+
+
+      var ewName ,vwName:string;
+      if (weighted!=0) {
+        fillInt(v_weight,1,1000);
+        //fillRandom(v_weight,0,100);
+        ewName = st.nextName();
+        vwName = st.nextName();
+        var vwEntry = new shared SymEntry(v_weight);
+        var ewEntry = new shared SymEntry(e_weight);
+        st.addEntry(vwName, vwEntry);
+        st.addEntry(ewName, ewEntry);
+      }
+      var srcName = st.nextName();
+      var dstName = st.nextName();
+      var startName = st.nextName();
+      var neiName = st.nextName();
+      var srcEntry = new shared SymEntry(src);
+      var dstEntry = new shared SymEntry(dst);
+      var startEntry = new shared SymEntry(start_i);
+      var neiEntry = new shared SymEntry(neighbour);
+      st.addEntry(srcName, srcEntry);
+      st.addEntry(dstName, dstEntry);
+      st.addEntry(startName, startEntry);
+      st.addEntry(neiName, neiEntry);
+      var sNv=Nv:string;
+      var sNe=Ne:string;
+      var sDirected=directed:string;
+      var sWeighted=weighted:string;
+
+      var srcNameR, dstNameR, startNameR, neiNameR:string;
+      if (directed!=0) {//for directed graph
+          if (weighted!=0) {// for weighted graph
+              repMsg =  sNv + '+ ' + sNe + '+ ' + sDirected + '+ ' + sWeighted +
+                    '+created ' + st.attrib(srcName)   + '+created ' + st.attrib(dstName) +
+                    '+created ' + st.attrib(startName) + '+created ' + st.attrib(neiName) +
+                    '+created ' + st.attrib(vwName)    + '+created ' + st.attrib(ewName);
+          } else {// for unweighted graph
+              repMsg =  sNv + '+ ' + sNe + '+ ' + sDirected + '+ ' + sWeighted +
+                    '+created ' + st.attrib(srcName)   + '+created ' + st.attrib(dstName) +
+                    '+created ' + st.attrib(startName) + '+created ' + st.attrib(neiName) ;
```

Comment:
Optimization:  Since the beginning of the result is the same this can be reduced to have the additional two pieces added on if `directed != 0`. i.e.
```chapel
repMsg = sNv + '+ ' + sNe + '+ ' + sDirected + '+ ' + sWeighted +
                    '+created ' + st.attrib(srcName)   + '+created ' + st.attrib(dstName) +
                    '+created ' + st.attrib(startName) + '+created ' + st.attrib(neiName);
if (weighted !=0 ) {
  repMsg += '+created ' + st.attrib(vwName) + '+created ' + st.attrib(ewName);
}
```

Reply:
Yes, you are correct!